docs/halibut-protocol.md

```diff
@@ -0,0 +1,83 @@
+# Halibut Protocol data exchange
```

Comment:
Consider mentioning that the protocol is inspired by JSON-RPC and adding a link to https://www.codeproject.com/Articles/554692/Halibut-a-secure-communication-stack-for-NET-Mono to provide some historical context.

Reply:
added